### PR TITLE
refactor(perception_ground_segmentation): rework parameters

### PR DESCRIPTION
**Pull request by karishma1911**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the perception_ground_segmentation

1. Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.

2.create schema

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by badai-nguyen — CHANGES_REQUESTED** [review]
@karishma1911 Thank you for you PR, 
IMO: since all default parameters of 3 filters will be removed, then those default parameters should be listed somewhere like `xx_ground_filter.param.yaml` in `config/`. Otherwise the current launch for specific scan_ground_filter (`scan_ground_filter.launch.py`) cannot be executed. 
We are sorry that default launch.py for remaining `ransac_ground_filter` and `ray_ground_filter` was not maintained well yet.

File: perception/ground_segmentation/src/ransac_ground_filter_nodelet.cpp [hunk 1]
```diff
@@ -85,18 +85,18 @@ using pointcloud_preprocessor::get_param;
...
-  voxel_size_z_ = declare_parameter("voxel_size_z", 0.04);
-  height_threshold_ = declare_parameter("height_threshold", 0.01);
-  debug_ = declare_parameter("debug", false);
+  base_frame_ = declare_parameter<std::string>("base_frame");
+  unit_axis_ = declare_parameter<std::string>("unit_axis");
+  max_iterations_ = declare_parameter<int>("max_iterations");
+  min_inliers_ = declare_parameter<int>("min_trial");
+  min_points_ = declare_parameter<int>("min_points");
+  outlier_threshold_ = declare_parameter<double>("outlier_threshold");
+  plane_slope_threshold_ = declare_parameter(<double>"plane_slope_threshold");
```
[the remark this answers is not in the document]
Comment (karishma1911) [reply perception/ground_segmentation/src/ransac_ground_filter_nodelet.cpp]: Thank you for suggestion...
I push the changes you mentioned above.

**Comment by badai-nguyen** [comment]
@karishma1911 
I have another comment.
Since some parameters use default values which only declared in `scan_ground_filter_nodelet.cpp` so `common_ground_filter` cannot started with your PR. One of the errors as below. Could you also add such missing parameters into [ground_segmentation.param.yaml](https://github.com/autowarefoundation/autoware_launch/blob/a13af90d620f63baa87b8920715c675f77371fda/autoware_launch/config/perception/obstacle_segmentation/ground_segmentation/ground_segmentation.param.yaml#L32) file also?
 
>Failed to load node 'common_ground_filter' of type 'ground_segmentation::ScanGroundFilterComponent' in container 'pointcloud_container': Component constructor threw an exception: Statically typed parameter 'radial_divider_angle_deg' must be initialized.

**Comment by kminoda** [comment]
@karishma1911 Hi, thank you for your contribution! What is the status of this PR? If this is stale, can we make this PR draft for now? (Of course feel free to re-open if it's ready for review)

**Comment by idorobotics** [comment]
@karishma1911 could you please provide updates?

**Comment by kminoda** [comment]
@karishma1911 Please fix the PR or recreate as another one :pray:

**Comment by kminoda** [comment]
@karishma1911 Let me convert this PR as a draft as it may be noisy for other developers. But feel free to reopen this PR once it is fixed. Thank you for your contribution!

**Comment by vividf** [comment]
I'll close this PR for now, as it doesn't appear to be actively progressing.

**Pull request closed without merging**